include/sys/util.h

```diff
@@ -97,6 +97,33 @@ constexpr size_t ARRAY_SIZE(T(&)[N]) { return N; }
 #define MIN(a, b) (((a) < (b)) ? (a) : (b))
 #endif
 
+/** @brief Return larger value of two provided expressions.
+ *
+ * Macro ensures that expressions are evaluated only once.
+ *
+ * @note Macro has limited usage compared to the standard macro as it cannot
+ *	 be used to generate constant integers or used outside a function.
+ */
```

Comment:
Your PR description says
```
Z_MAX and Z_MIN cannot be used in cases like:

    constant integer generation (e.g. __aligned(Z_MAX(4,5))
    static variables (e.g. array for stack static u8_t array[Z_MAX(...)];)
```
Why not add these comments also here to make it more clear when the macros cannot be used.